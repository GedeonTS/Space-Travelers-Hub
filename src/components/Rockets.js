import { useDispatch, useSelector } from 'react-redux';
import { reserveRocket } from '../redux/rockets/rockets';
import Rocket from './rocket';

const Rockets = () => {
  const rockets = useSelector((state) => state.rocketList);

  const dispatch = useDispatch();
  const handleReservedRocket = (id) => {
    dispatch(reserveRocket(id));
  };

  return (
    <div>
      {rockets.map((rocket) => (
        <Rocket
          key={rocket.id}
          name={rocket.name}
          description={rocket.description}
          flickrImages={rocket.flickr_images}
          reserved={rocket.reserved}
          reservedRocket={() => handleReservedRocket(rocket.id)}
        />
      ))}
    </div>
  );
};

export default Rockets;
