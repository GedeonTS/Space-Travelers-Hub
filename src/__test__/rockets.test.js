import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configStore';
import Rockets from '../components/rocket';

describe('display rockets component', () => {
  test('test if rocket component renders in the dom', () => {
    render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    expect(screen.queryByText(/Reserve Rocket/)).toBeTruthy();
    expect(screen.queryByText(/Mission/)).toBeNull();
  });
  test('snapshot renders properly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Rockets />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
