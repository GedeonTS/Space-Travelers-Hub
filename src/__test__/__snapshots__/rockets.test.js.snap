// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display rockets component snapshot renders properly 1`] = `
<div
  className="rocketContainer"
>
  <img
    alt="rocket"
    className="imgContainer"
    src=""
  />
  <div
    className="displayInfo"
  >
    <h1
      className="headerText"
    />
    <p
      className="descriptionTxt"
    />
    <button
      className="buttonStyle"
      onClick={null}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;
